src/WebJobs.Extensions.OpenAI/Assistants/AssistantService.cs

```diff
@@ -364,6 +365,27 @@ public async Task PostMessageAsync(AssistantPostRequest request, CancellationTok
 
         // Add the batch of table transaction actions to the table
         await this.tableClient.SubmitTransactionAsync(batch, cancellationToken);
+
+        // return the latest message in the chat state
+        List<ChatMessageTableEntity> filteredChatMessages = chatState.Messages
+            .Where(msg => msg.CreatedAt > timeFilter)
+            .ToList();
```

Comment:
We should only be returning a single message, not multiple, and it should be the very last message (which is expected to be an assistant message). We don't want to return any of the intermediate function call messages since those are not useful for the end user.

Reply:
updated with the same